From dce69cadd858bd959dd41edf82860cd370ca1fa8 Mon Sep 17 00:00:00 2001
From: Karthik Ramanan <a0393906@ti.com>
Date: Fri, 14 Mar 2014 18:55:12 +0530
Subject: [PATCH] weston-drm: Enable multiple displays

We are currently seeing an issue with missing VBlanks for LCD and
HDMI connectors which leads to display not getting refreshed.
This patch can be considered as a workaround.

Signed-off-by: Karthik Ramanan <a0393906@ti.com>
---
 src/compositor-drm.c |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/compositor-drm.c b/src/compositor-drm.c
index 79dcc6c..ef2fbd2 100644
--- a/src/compositor-drm.c
+++ b/src/compositor-drm.c
@@ -709,6 +709,7 @@ page_flip_handler(int fd, unsigned int frame,
 {
 	struct drm_output *output = (struct drm_output *) data;
 	uint32_t msecs;
+	uint32_t bail;
 
 	/* We don't set page_flip_pending on start_repaint_loop, in that case
 	 * we just want to page flip to the current buffer to get an accurate
@@ -721,7 +722,12 @@ page_flip_handler(int fd, unsigned int frame,
 
 	output->page_flip_pending = 0;
 
-	if (!output->vblank_pending) {
+	if (output->vblank_pending) {
+		weston_log("VBlank is pending for connector = %d, frame = %d\n", output->connector_id, frame);
+		bail = 1;
+	}
+
+	if (!output->vblank_pending || bail) {
 		msecs = sec * 1000 + usec / 1000;
 		weston_output_finish_frame(&output->base, msecs);
 
-- 
1.7.9.5

